themes/wporg-translate-events-2024/blocks/components/event/date.php

```diff
@@ -0,0 +1,24 @@
+<?php namespace Wporg\TranslationEvents\Theme_2024;
+use Wporg\TranslationEvents\Translation_Events;
+
+register_block_type(
+	'wporg-translate-events-2024/components-event-start',
+	array(
+		// phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.Found
+		'render_callback' => function ( array $attributes ) {
```

Comment:
I think each block should only get as attribute the data that it requires. So instead of the block getting an event id and then using that id to get the event from the database, and then retrieving the data it cares about from the event, the block would instead get the data it needs directly as an attribute. In this case, the block would get the start datetime, instead of the event id.

The responsibility of this block is to render the start datetime of the event, but the block itself should not need to know that events are stored in the database. For example, you should be able to call this block with an instance of an event that is already in memory.

I think applying this principle will simplify things quite a bit, for example, we probably won't need to add caching to the repository because the blocks won't need to interact with the database.

This would apply to all blocks we're adding, not just this one.

Reply:
I agree, thank you for this.